src/http/common.cr

```diff
@@ -64,11 +67,45 @@ module HTTP
 
         yield headers, body
         break
+      else # HeaderLine
+        headers_size += header_line.bytesize
+        break if headers_size > MAX_HEADER_SIZE
+
+        break unless headers.add?(header_line.name, header_line.value)
       end
+    end
+  end
+
+  private def self.read_header_line(io) : HeaderLine | EndOfRequest | Nil
+    # Optimization: check if we have a peek buffer
+    if peek = io.peek
+      # peek.empty? means EOF (so bad request)
+      return nil if peek.empty?
+
+      # Find \r\n (first \n, then \r before it)
```

Comment:
Previously we allowed just `\n` to end a header line. Some bad clients may well send just `\n`. They should probably be supported. NGINX supports just `\n`: https://github.com/nginx/nginx/blob/4bf4650f2f10f7bbacfe7a33da744f18951d416d/src/http/ngx_http_parse.c#L770-L777.

Reply:
This actually applies to the request line parsing too...

Reply:
When the request line is parsed from an IO that can't peek, it already accepts `\n` because `IO#gets` reads the `\n` optionally. It should also be optional for the peek branch and header parser.

Reply:
Thanks for pointing out that this does fall back to the old `gets` code if `\r\n` isn't used so it's not a regression!

Reply:
Yes, it falls back, but we could treat that case directly in the peek branch.

Reply:
Thanks! Fixed in #8009